infra/database/postgis.yml

```diff
@@ -0,0 +1,98 @@
+---
+
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: zgw
+
+---
+
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: db-pvc
+  namespace: zgw
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 20Gi
+
+---
+
+# Deployment
+
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  name: postgis
+  namespace: zgw
+  labels:
+    k8s-app: postgis
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: postgis
+  template:
+    metadata:
+      name: postgis
+      labels:
+        k8s-app: postgis
+    spec:
+      volumes:
+        - name: storage
+          persistentVolumeClaim:
+           claimName: db-pvc
+        - name: postgis-initdb
+          configMap:
+            name: postgis-initdb-config
+
+      containers:
+        - name: postgres
+          image: mdillon/postgis:11
+          env:
+            - name: POSTGRES_PASSWORD
+              value: <REDACTED>
+          volumeMounts:
+            - name: storage
+              mountPath: "/var/lib/postgresql/data"
+              subPath: zgw/database
+            - name: postgis-initdb
+              mountPath: /docker-entrypoint-initdb.d
+
+---
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: postgis-initdb-config
+  namespace: zgw
+data:
+  initdb.sql: |
+    CREATE DATABASE "zrc";
```

Comment:
Is this actually 'run once' or could it be hitting these create's again after a pod deletion / recreation? Would create database if not exists be helpful?

Reply:
it is only run once, when the DB cluster is initializing. Which is problematic when a new database is added to an existing cluster - this is currently a manual action.

We could probably let our containers do a `CREATE DATABASE foo;` as well, as long as the db user has sufficient privileges.